rpcs3/rpcs3qt/pad_led_settings_dialog.cpp

```diff
@@ -0,0 +1,107 @@
+﻿#include "pad_led_settings_dialog.h"
+#include <QColorDialog>
+
+pad_led_settings_dialog::pad_led_settings_dialog(const int& colorR, const int& colorG, const int& colorB, const bool& led_low_battery_blink, const bool& led_battery_indicator, const int& led_battery_indicator_brightness, QDialog * parent)
+    : QDialog(parent)
+    , ui(new Ui::pad_led_settings_dialog)
+    , m_initial{colorR, colorG, colorB, led_low_battery_blink, led_battery_indicator, led_battery_indicator_brightness}
+{
+	ui->setupUi(this);
+	setFixedSize(size());
+	m_new = m_initial;
+
+	// Fill the form
+	ui->hs_indicator_brightness->setValue(m_new.battery_indicator_brightness);
+	ui->cb_led_blink->setChecked(m_new.low_battery_blink);
+	ui->cb_led_indicate->setChecked(m_new.battery_indicator);
+	switch_groupboxes(m_new.battery_indicator);
+	update_slider_label(m_new.battery_indicator_brightness);
+	connect(ui->bb_dialog_buttons, &QDialogButtonBox::clicked, [this](QAbstractButton* button) {
+		read_form_values();
```

Comment:
Why is this always called?
And where is the apply button handled?

Reply:
`read_form_values` fills `m_new` with current states of checkboxes and the slider. An alternative to this is to connect to events on each of them. Would that be a better way to go about it?

Apply button doesn't need explicit handling since we pass the data to `pad_settings_dialog` regardless of which button is clicked, while OK and Cancel do require additional handling.

Reply:
You don't need to update m_new twice with cancel though.

Reply:
With cancel m_new is reassigned to the original values, this way it passes the old settings back to `pad_settings_dialog` which cancels any changes that might have been passed if the user clicked Apply.

Reply:
I know. I was referring to read_form_values

Reply:
Oh, my bad, makes sense. I'll call `read_form_values` for OK and Apply only.